src/AutoRest.CSharp/Mgmt/Generation/MgmtLongRunningOperationWriter.cs

```diff
@@ -2,85 +2,78 @@
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
 using System;
-using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
 using AutoRest.CSharp.Generation.Writers;
 using AutoRest.CSharp.Mgmt.AutoRest;
-using AutoRest.CSharp.Output.Models.Types;
+using AutoRest.CSharp.Mgmt.Output;
 using Azure;
-using Azure.Core;
-using Azure.Core.Pipeline;
-using Azure.ResourceManager;
-using Request = Azure.Core.Request;
 
 namespace AutoRest.CSharp.Mgmt.Generation
 {
     internal class MgmtLongRunningOperationWriter
     {
         private readonly CodeWriter _writer;
-        private readonly string _name;
-        private readonly string _genericString;
-        private readonly bool _isGeneric;
-        private readonly string _waitMethod;
-        private readonly Type _operationType;
-        private readonly Type _responseType;
-        private readonly Type _operationOrResponseType;
-        private readonly FormattableString _operationSourceString;
-        private readonly string _sourceString;
 
+        private MgmtLongRunningOperation _lro;
         public string Filename { get; }
 
-        public MgmtLongRunningOperationWriter(bool isGeneric)
+        public MgmtLongRunningOperationWriter(MgmtLongRunningOperation longRunningOperations)
         {
             _writer = new CodeWriter();
-            _isGeneric = isGeneric;
-            _name = $"{MgmtContext.Context.DefaultNamespace.Split('.').Last()}ArmOperation";
-            _genericString = isGeneric ? "<T>" : string.Empty;
-            Filename = isGeneric ? $"LongRunningOperation/{_name}OfT.cs" : $"LongRunningOperation/{_name}.cs";
-            _waitMethod = isGeneric ? "WaitForCompletionAsync" : "WaitForCompletionResponseAsync";
-            _operationType = isGeneric ? typeof(ArmOperation<>) : typeof(ArmOperation);
-            _responseType = isGeneric ? typeof(Response<>) : typeof(Response);
-            _operationOrResponseType = isGeneric ? typeof(OperationOrResponseInternals<>) : typeof(OperationOrResponseInternals);
-            _operationSourceString = isGeneric ? (FormattableString)$"{typeof(IOperationSource<>)} source, " : (FormattableString)$"";
-            _sourceString = isGeneric ? "source, " : string.Empty;
+            _lro = longRunningOperations;
+            Filename = _lro.IsGeneric ? $"{_lro.Type.Name}OfT.cs" : $"{_lro.Type.Name}.cs";
         }
 
         public void Write()
         {
             using (_writer.Namespace(MgmtContext.Context.DefaultNamespace))
             {
                 _writer.Line($"#pragma warning disable SA1649 // File name should match first type name");
-                _writer.Line($"internal class {_name}{_genericString} : {_operationType}");
+                var genericString = _lro.IsGeneric ? "<T>" : string.Empty;
+                _writer.Line($"internal class {_lro.Type.Name}{genericString} : {_lro.BaseType}");
```

Comment:
If we introduce a type for LRO's, there should be no type manipulations in the writer. This should be something like 
```cs
using (writer.Scope($"internal partial class {_lro.Type:D}")) {
```

Right now `CodeWriter` doesn't support `D` for the type, I'll add it in separate PR

Reply:
Shall we also introduce something in `CSharpType` to represent a generic type with a generic type argument placeholder? 
So that this could write to `SampleArmOperations<T>`

Reply:
`CSharpType` can represent a generic type, I've updated CodeWriter so that it can correctly write it: https://github.com/Azure/autorest.csharp/pull/1994

Reply:
Thank you! But currently we are still missing a piece that `TypeProvider` never generate its `Type` as a generic type: 
https://github.com/Azure/autorest.csharp/blob/2e5b4431de73126a7f4b493bfe25cd5c6a9e824e/src/AutoRest.CSharp/Common/Output/Models/Types/TypeProvider.cs#L22
And we might rarely have a case that we need to write a declaration of a framework type. Should we also have this fixed in a separate PR?

Reply:
@AlexanderSher I added some attempt of doing that in [the latest commit](https://github.com/Azure/autorest.csharp/pull/1980/commits/9f865a65d47ff5472796654764c477ea20da14e8), please take a look.

And BTW do we have a handy way to instantiate a generic type placeholder (like `T`) of type `CSharpType`? I have to use the generic type parameter from a system type, and then convert them to CSharpType.